clients/src/main/java/org/apache/kafka/clients/consumer/internals/TopicMetadataRequestManager.java

```diff
@@ -210,7 +194,7 @@ private NetworkClientDelegate.UnsentRequest createUnsentRequest(
         private void handleError(final Throwable exception,
                                  final long completionTimeMs) {
             if (exception instanceof RetriableException) {
-                if (completionTimeMs >= expirationTimeMs) {
+                if (isExpired()) {
```

Comment:
this logic of what we want to do on retriable errors seems to be the same for all retriable errors (no matter if found in `handleError` or `handleResponse`). Could we maybe consolidate it in a single place `maybeExpireOnRetriable` or something like it, that would be consistently applied whenever we need to maybe expire?

Reply:
I will take a look at refactoring this.

Reply:
I refactored the code so that `handleResponse()` calls `handleError()` if it hits an error.